File: frontend/src/api/graph.ts
import { post } from './client';

export interface GraphNode {
  id: string;
  label: string;
  properties: Record<string, any>;
}

export interface GraphEdge {
  source_id: string;
  target_id: string;
  type: string;
  properties: Record<string, any>;
}

export interface GraphNeighborhood {
  center_node_id: string;
  nodes: GraphNode[];
  edges: GraphEdge[];
}

export interface NeighborhoodRequest {
  node_id: string;
  label: string;
  depth?: number;
}

export const graphApi = {
  async getNeighborhood(request: NeighborhoodRequest): Promise<GraphNeighborhood> {
    return post<GraphNeighborhood>('/api/v1/graph/neighborhood', request);
  },
};

File: backend/app/schemas/graph.py
from __future__ import annotations
from pydantic import BaseModel, Field
from typing import Any, Dict, List, Optional


class GraphNode(BaseModel):
    id: str
    label: str
    properties: Dict[str, Any] = Field(default_factory=dict)


class GraphEdge(BaseModel):
    source_id: str
    target_id: str
    type: str
    properties: Dict[str, Any] = Field(default_factory=dict)


class GraphNeighborhood(BaseModel):
    center_node_id: str
    nodes: List[GraphNode]
    edges: List[GraphEdge]


class NeighborhoodRequest(BaseModel):
    node_id: str
    label: str
    depth: Optional[int] = 1
